content/doc/developer/development-environment/index.adoc

```diff
@@ -1,24 +1,34 @@
 ---
 layout: developerchapter
-wip: true
 summary: Preparing your development environment for Jenkins core and plugin development
 references:
-- url: https://wiki.jenkins.io/display/JENKINS/IntelliJ+IDEA+plugin+for+Stapler
-  title: IntelliJ IDEA plugin for Stapler
-- url: https://wiki.jenkins.io/display/JENKINS/NetBeans+plugin+for+Stapler
-  title: NetBeans plugin for Stapler
-- url: https://wiki.jenkins.io/display/JENKINS/Setting+up+Eclipse+to+build+Jenkins
-  title: Setting up Eclipse to build Jenkins
-- url: https://wiki.jenkins.io/display/JENKINS/Jenkins+plugin+development+with+Eclipse
-  title: Jenkins plugin development with Eclipse
-- url: https://wiki.jenkins.io/display/JENKINS/Eclipse+alternative+build+setup
-  title: Eclipse alternative build setup
 - url: https://wiki.jenkins.io/display/JENKINS/Developing+with+JRebel
   title: Developing with JRebel # TODO find someone who can write actual documentation for this
-- url: https://wiki.jenkins.io/display/JENKINS/Using+Artifactory+for+Configuring+a+Jenkins+Build+Environment
-  title: Using Artifactory for Configuring a Jenkins Build Environment # TODO really?
-- url: https://wiki.jenkins.io/display/JENKINS/Jenkins+Development+Environment+with+Nexus
-  title: Jenkins Development Environment with Nexus # TODO really?
 ---
 
 = Setting up a Development Environment
+
+== IntelliJ
+https://plugins.jetbrains.com/plugin/1885-stapler-plugin-for-intellij-idea
+
+This IDEA plugin makes it easy to develop Jenkins and its plugins on IntelliJ IDEA
+
+This plugin has the following features:
+
+* Cmd+B navigation from the Jelly tags to their definitions
+* Error checks and auto completion on attributes and elements of taglibs; also see Writing Jelly views with IDE assistance
+* Syntax error checks on JEXL expressions
+* "Go to stapler view" to jump from a Java class to its views (Cmd+Shift+P)
+* Select a string expression, then "Refactor" > "i18n for Stapler" to create a message resource
+
+== Netbeans
+http://plugins.netbeans.org/plugin/43938/?show=true
+
+== XSD configuration
+If there isn't a plugin available for your IDE you can manually add the taglib to your IDE.
+
+XML schemas for tag libraries defined in Jelly, stapler, and Jenkins are available. Most modern IDEs and other XML editors can use them to provide auto-completion, which improves your productivity.
+
+For link:https://commons.apache.org/jelly/libs/index.html[Jelly core, define, format, util, and other tag libraries maintained in Apache], schemas are available in https://github.com/kohsuke/maven-jellydoc-plugin/tree/master/maven-jellydoc-plugin/schemas. These schemas rarely change.
+For stapler, schema is available in https://stapler.kohsuke.org/taglib.xsd. New tags are occasionally added here.
+For taglibs defined in Jenkins, schemas are .
```

Comment:
Todo: looks like I missed adding the link here, will fix up

Reply:
The taglibs are in the sidebar of https://jenkins.io/doc/developer/book/ -- do we need a standalone page similar to https://jenkins.io/doc/developer/javadoc/ to limit redundancy?

Reply:
Yes I think so, will make the change